sql/table.cc

```diff
@@ -10116,6 +10117,15 @@ bool TABLE_LIST::init_derived(THD *thd, bool init_view)
     bool forced_no_merge_for_update_delete=
            belong_to_view ? belong_to_view->updating :
                            !unit->outer_select()->outer_select();
+
```

Comment:
This got me wondering, when how come table hints could be adjusted already?  The testcase shows this happens when merge operations moves derived tables from one select to another.  Please add a comment.